pkg/sql/planner.go

```diff
@@ -343,19 +343,17 @@ func (p *planner) exec(ctx context.Context, sql string, args ...interface{}) (in
 	return countRowsAffected(params, plan)
 }
 
-func (p *planner) fillFKTableMap(ctx context.Context, m sqlbase.TableLookupsByID) error {
-	for tableID := range m {
-		table, err := p.session.tables.getTableVersionByID(ctx, p.txn, tableID)
+func (p *planner) lookupFKTable(
+	ctx context.Context, tableID sqlbase.ID,
```

Comment:
somewhere (maybe here?) do we also need to check that the user has the appropriate privileges on the other tables? probably also want a logic test that switches users to observe that check?